### PR TITLE
Update dependencies to latest version

### PR DESCRIPTION
**Pull request by arunkumarchacko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -84,24 +84,24 @@
...
-    <google.auth.version>1.7.0</google.auth.version>
+    <google.api-client.version>2.0.1</google.api-client.version>
+    <google.api-client-libraries.version>2.0.0</google.api-client-libraries.version>
+    <google.api-bigquery.version>v2-rev20221028-${google.api-client-libraries.version}</google.api-bigquery.version>
+    <google.api-iamcredentials.version>v1-rev20211203-${google.api-client-libraries.version}</google.api-iamcredentials.version>
+    <google.api-storage.version>v1-rev20220705-${google.api-client-libraries.version}</google.api-storage.version>
+    <google.auth.version>1.12.1</google.auth.version>
     <google.auto-value.version>1.8.2</google.auto-value.version>
-    <google.cloud-bigquerystorage.version>1.22.5</google.cloud-bigquerystorage.version>
+    <google.cloud-bigquerystorage.version>2.25.0</google.cloud-bigquerystorage.version>
```
[the remark this answers is not in the document]
Comment (arunkumarchacko) [reply pom.xml:94]: LTS is using this version. If there is some issue, our tests should catch it.

[the remark this answers is not in the document]
Comment (blakeli0) [reply pom.xml:94]: FYI, these are the release notes for [google-api-java-client](https://github.com/googleapis/java-bigquerystorage/releases/tag/v2.0.0) and [bigquerystorage](https://github.com/googleapis/java-bigquerystorage/releases/tag/v2.0.0), which I believe have very low risks. If all the tests are passing, you should be fine. 
We usually keep all libraries in LTS up to date, so it's important to keep the dependencies used by LTS up to date in this repo as well, or at least make sure all dependencies used by LTS are compatible.

**Review by Deependra-Patel — APPROVED** [review]

**Pull request merged**